src/Microsoft.PowerShell.SHiPS/SHiPSScriptRunner.cs

```diff
@@ -86,8 +86,14 @@ internal class PSScriptRunner
 
                 if (errors.Count > 0)
                 {
+                    //remove the cached child nodes for the failed node
+                    node.Children?.Clear();
+
+                    //remove the node from its parent's children list so that it won't show again when a user types dir -force
+                    node.Parent?.Children.RemoveSafe(node.Name);
```

Comment:
if Parent is null, then won't Children be null too? So you get a nullref anyway

Reply:
SHiPSDirectory class for folders, so it has Children list created during initiation.

Reply:
If `Parent` is null, the operator terminates execution of that line of code, so it is safe.